[YANG Mingyang] iP

myangat0343 opened the pull request:
> ### Duke
> > Try to Manage your study and life : )
> > 
> 
> * text-based
> * easy to learn
> * convenient to use 
> 
> All you need to do is:
> 
> 1. Download from [here](https://github.com/myangat0343/ip/releases/tag/v0.1)
> 2. Double click it
> 3. Follow the instructions
> 4. Enjoy your time with it.
> 
> And it's **FREE** 🎉
> 
> Functionalities:
> 
> * [x]  View all tasks 
> * [x]  Managing tasks
> * [x]  Load your tasks
> * [x]  Save your task
> * [x]  Add more Users
> * [x]  Set up GUI
> 
> If you are a java programmer, you can also use it for practice. Here is the `main` method.
> 
> ```java
> public static void main(String[] args) {
>         Duke duke = new Duke();
>         duke.launch();
>     }
> ```

Commit:
Add Gradle support

Myangat committed:
For Level 2, I did it first, cause I haven't seen the Level 1

Myangat committed:
Level 4

Myangat committed:
Duke A-TextUiTesting: Automated Text UI Testing, I change the expected output file

Myangat committed:
Level 5: Only finish the minimal goal

Myangat committed:
Level 6: Delete

Myangat committed:
Level 7: Enumerations: currently not applicable?

Myangat committed:
Level-7

Myangat committed:
Level 8

Myangat committed:
ignore the file

Myangat committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Level-7

# Conflicts:
#	src/main/java/Duke.java

Myangat committed:
System.getProperty("user.home"); read files correctly

Myangat committed:
A-MoreOOP: difficult

Myangat committed:
A-Packages

Myangat committed:
Merge branch 'add-gradle-support'
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

A-Gradle

lizekai-richard commented on src/main/java/Storage/Storage.java:
```diff
@@ -0,0 +1,110 @@
+                    throw new Exception();
+            }
+
+        } catch (Exception e)
+        {
+            System.out.println("StoredFiledException!");
+            throw new StoredFileException();
+        }
```
Perhaps you should follow the Egyptian style here for your brackets 😜

myangat0343 replied:
Haha thanks for comments~

lizekai-richard reviewed (commented):
High readability! 👍

Myangat committed:
Set up the Duck Package

Myangat committed:
Add JUnit Tests

Myangat committed:
Update More-OOP

Myangat committed:
Fix some bugs

Myangat committed:
Remove test folder in order to re-do this part

Myangat committed:
Done for the JUnit

Myangat committed:
Working on JAR

Myangat committed:
Finish the JAR, but running has IO Exception e

Myangat committed:
ignore the jar file and not commit it

Myangat committed:
Merge branch 'master' into A-JUit

* master:
  Finish the JAR, but running has IO Exception e
  Working on JAR

Myangat committed:
Fix some bugs

SweetPotato0213 commented on src/main/java/Duke.java:
```diff
@@ -10,16 +10,70 @@ public static void main(String[] args) {
         String input;
         input = sc.nextLine();
-        List<String> list = new ArrayList<>();
+        List<Task> taskList = new ArrayList<>();
```
Maybe can use tasks here instead of taskList?
```suggestion
        List<Task> tasks = new ArrayList<>();
```

SweetPotato0213 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,20 @@
+public abstract class Task {
+    protected String discription;
+    protected boolean isDone;
```
Good boolean naming

SweetPotato0213 commented on src/main/java/Duke.java:
```diff
@@ -10,16 +10,70 @@ public static void main(String[] args) {
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  " + todo);
+                    System.out.println("Now you have " + num + " tasks in the list");
+
```
Maybe this empty line can be removed?

myangat0343 replied:
Thanks for your suggestions!

SweetPotato0213 reviewed (commented):
LGTM!

Myangat committed:
Improve FileReader class with two methods convertToTask() and load()

Myangat committed:
Add Parse

Myangat committed:
Add JavaDoc

Myangat committed:
Add the class FindTaskCommand

Myangat committed:
Update class Tasks, ToDo, Deadline, Events ...

ryanczx commented on src/main/java/Duke/Storage/FileReader.java:
```diff
@@ -0,0 +1,103 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.Scanner;
```
Neat and easy to read ordering of package and import statements!

ryanczx commented on src/main/java/Duke/Storage/FileReader.java:
```diff
@@ -0,0 +1,103 @@
+
+                default:
+                    throw new Exception();
+            }
```
Minor issue no worries! However the switch statements style is different for this module.

I got the following from the module website.

```
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```

myangat0343 replied:
Thanks so much!

ryanczx reviewed (commented):
Looks great overall!

Myangat committed:
Improve the class TaskList (partially)

Myangat committed:
Improve the classes for Commands.

Myangat committed:
Improve the class parser to handle input commands

Myangat committed:
Duke can read from stored file

Myangat committed:
Level-10: GUI, JavaFX

Myangat committed:
Level-10 Solve the bug (use .png)

Myangat committed:
Handle both Duke.class and Main.class

Myangat committed:
Some improvement

Myangat committed:
no message

Myangat committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Handle both Duke.class and Main.class

Myangat committed:
ip Week 5: Generate a new JAR file

Myangat committed:
correct the style, (switch..case..)

Myangat committed:
no message

Myangat committed:
no message

Myangat committed:
Import sort commmand

Myangat committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Import sort commmand
  no message

# Conflicts:
#	src/main/java/Duke/Duke.java
#	src/main/java/Duke/Parser/CLIParser.java
#	src/main/java/Duke/Storage/FileReader.java

Myangat committed:
Save Function works

Myangat committed:
Adds more comments

Myangat committed:
Improves GUI performance

Myangat committed:
Fix some bugs

Myangat committed:
Change some property of DialogBox

Myangat committed:
Add more JUnit Tests

Adding JUnit tests can be beneficial to testing the behaviour of code.

GUIParser is an important class to handle input String.

Using Assertion(expected, actual) to check if the code works.

Myangat committed:
Improve a JUnit Test

Improving JUnit tests can be beneficial to testing the behaviour of code.

TaskList is an important class to do operation on task list.

Using Assertion(expected, actual) to check if the code works.

Myangat committed:
Use Assert

Use assert feature to document important assumptions that should hold at
various points in the code.

Myangat committed:
Add more Assertions

Myangat committed:
Improve code quality.

myangat0343 committed:
Merge branch 'master' into branch-A-CodeQuality

myangat0343 committed:
Merge pull request #2 from myangat0343/branch-A-CodeQuality

Improve code quality

Myangat committed:
Add Ui.png

Myangat committed:
Finish UG

Myangat committed:
Improve code quality

Names representing packages should be in all lower case.

Every class should be part of some package.

It will help you and other developers easily understand the code base
when all the classes have been grouped in packages.

Myangat committed:
Improve code quality

myangat0343 committed:
Merge pull request #3 from myangat0343/branch-A-CodeQuality

Branch a code quality

Myangat committed:
Delete some file

Avoid committing binary files (e.g., *.class, *.jar, *.exe) or
third-party library files in to the repo.

myangat0343 committed:
Delete src/test/test/test directory

Avoid committing binary files (e.g., *.class, *.jar, *.exe) or third-party library files into the repo.

Myangat committed:
Avoid assert to generate exception message

Myangat committed:
Fix some bugs
